### PR TITLE
feat: platform literals

### PR DESCRIPTION
**Pull request by baszalmstra**
Modifies the schema to:

- Turns platform strings into literals. 
- Move schema tasks to seperate environment.

### CODE REVIEW

File: .github/workflows/schema.yml [hunk 1]
```diff
@@ -28,4 +28,4 @@ jobs:
```
Comment (pavelzw) [inline .github/workflows/schema.yml:28]: image
this should be a 

```yml
- uses: prefix-dev/setup-pixi
  with:
    pixi-version: v0.17.1
    environments: schema
```

File: .github/workflows/schema.yml [hunk 2]
```diff
@@ -28,4 +28,4 @@ jobs:
 
         - name: Test Schema
           run: |
-            pixi run -e default test-schema
+            pixi run -e schema test-schema
```
Comment (pavelzw) [inline .github/workflows/schema.yml:31]: How about also running fmt-schema + generate-schema in CI and checking the git diff?

File: .github/workflows/schema.yml (hunk 1 above)
Comment (ruben-arts) [reply .github/workflows/schema.yml:28]: I actually think we should dog-food ourselfs more so we should always use latest now you point this out.

File: .github/workflows/schema.yml (hunk 2 above)
Comment (ruben-arts) [reply .github/workflows/schema.yml:31]: That is actually a good idea!

Comment (pavelzw) [reply .github/workflows/schema.yml:31]: https://github.com/prefix-dev/recipe-format/blob/ba04de9f7dd03573f979ffa80413c1e814cb669c/.github/workflows/ci.yml#L16-L30 😄  while we're at it we can also add ruff linting

Comment (baszalmstra) [reply .github/workflows/schema.yml:42]: Done!

File: .github/workflows/schema.yml [hunk 3]
```diff
@@ -25,7 +25,20 @@ jobs:
...
+
+        - name: Ensure schema is up-to-date
+          run: |
+            pixi run -e schema generate-schema
+            git diff --exit-code
 
         - name: Test Schema
           run: |
-            pixi run -e default test-schema
+            pixi run -e schema lint-schema
```
Comment (ruben-arts) [inline .github/workflows/schema.yml:37]: Linting is done by the pre-commit stage
```suggestion
```

File: .github/workflows/schema.yml [hunk 4]
```diff
@@ -25,7 +25,20 @@ jobs:
...
+            pixi run -e schema test-schema
+
+        - name: Check formatting
+          run: |
+            pixi run -e schema test-schema
+            git diff --exit-code
```
Comment (ruben-arts) [inline .github/workflows/schema.yml:43]:
```suggestion
            pixi run -e schema test-schema
            git diff --exit-code
```
This only has to be run once

File: .github/workflows/schema.yml [hunk 5]
```diff
@@ -25,7 +25,20 @@ jobs:
           with:
             pixi-version: v0.15.2
```
Comment (ruben-arts) [inline .github/workflows/schema.yml:26]:
```suggestion
            pixi-version: latest
```

Comment (pavelzw) [reply .github/workflows/schema.yml:26]: You can remove this line altogether as `latest` is the default. Also, `cache: true` is also the default.

**Review by ruben-arts — APPROVED** [review]

**Pull request merged**